src/storage/src/source/postgres/replication.rs

```diff
@@ -326,6 +326,18 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
             upper_cap_set.downgrade([&resume_lsn]);
             trace!(%id, "timely-{worker_id} replication reader started lsn={resume_lsn}");
 
+            // Emitting an initial probe before we start waiting for rewinds ensures that we will
+            // have a timestamp binding in the remap collection while the snapshot is processed.
+            // This is important because otherwise the snapshot updates would need to be buffered
+            // in the reclock operator, instead of being spilled to S3 in the persist sink.
+            let probe_ts = (config.now_fn)().into();
+            let max_lsn = super::fetch_max_lsn(&*metadata_client).await?;
+            // We can only produce the upstream table state accurately for LSN's >= the
+            // `resume_lsn`, so our first probe frontier must not be before `resume_lsn`.
+            let max_lsn = std::cmp::max(max_lsn, resume_lsn);
+            let probe = Probe { probe_ts, upstream_frontier: Antichain::from_elem(max_lsn) };
+            probe_output.give(&probe_cap[0], probe);
```

Comment:
This seems to work, but I have some uncertainty about whether `max(max_lsn, resume_lsn)` is the correct `upstream_frontier` to use here. We have to use something >= `slot_metadata.confirmed_flush_lsn`, and we _may_ have to use something >= `resume_lsn`, not sure. Also fetching the `max_lsn` seems like the right thing to do, to respect the probe semantics, but I'm not sure how important it is.

Reply:
I think using the max lsn unmodified is fine as long as it is fetched after the slot has been created (which it is, here). After creating a slot we can always recreate the state of the max LSN of the database. I don't think it can ever happen that `resume_lsn` is higher than a fresh `max_lsn`

Reply:
Put in a soft assert now instead.